src/player.js

```diff
@@ -1,204 +1,59 @@
-//TODO: separate out the canvas logic and the hevc decoder logic
-const YUVBuffer = require('yuv-buffer')
-const YUVCanvas = require('yuv-canvas')
+const InitMp4Parser = require('./demuxer/mp4')
 const Module = require('./missile.js')
-const AudioImp = require('./decoder/audio')
-const def = require('./consts')
-module.exports = config => {
+const Audio = require('./decoder/audio')
+const decodeVideoFrame = require('./decoder/video')
+const canvas = require('./canvas')
+module.exports = ({container, url}) => {
+    const videoFrames = []
+    const audioFrames = []
     const player = {
-        config: {
-            width: config.width || def.DEFAULT_WIDTH,
-            height: config.height || def.DEFAULT_HEIGHT,
-            fps: config.fps || def.DEFAULT_FPS,
-            fixed: config.fixed || def.DEFAULT_FIXED,
-            sampleRate: config.sampleRate || def.DEFAULT_SAMPLERATE,
-            appendHevcType: config.appendHevcType || def.APPEND_TYPE_STREAM
-        },
-        frameList: [],
-        stream: new Uint8Array(),
-        audio: new AudioImp(),
-        durationMs: -1.0,
-        videoPTS: -1,
         loop: null,
-        isPlaying: false
-    }
-    player.setSize = (width, height) => {
-        player.config.width = width || def.DEFAULT_WIDTH
-        player.config.height = height || def.DEFAULT_HEIGHT
-    }
-    player.setFrameRate = (fps) => {
-        player.config.fps = fps || def.DEFAULT_FPS
-    }
-    player.setDurationMs = (durationMs = -1) => {
-        player.durationMs = durationMs
-        player.audio.setDurationMs(durationMs)
-    }
-    player.appendAACFrame = streamBytes => player.audio.addSample(streamBytes)
-    player.endAudio = () => {
-        player.audio.stop()
-        player.audio.init({
-            sampleRate  : player.config.sampleRate,
-            appendType  : player.config.appendHevcType
-        })
-    }
-    player.cleanSample = () => player.audio.cleanQueue()
-    player.pause = () => {
-        player.loop && window.clearInterval(player.loop)
-        player.audio.pause()
-    }
-    player.play = callback => {
-        player.loop = window.setInterval(() => {
-            player.videoPTS * 1000 >= player.durationMs && player.stop()
-            player.playFrame()
-            callback && callback(player.videoPTS)
-        }, 1000 / player.config.fps)
-        player.audio.play()
-        console.log('Playing ...')
-    }
-    player.stop = () => {
-        console.log("============ STOP ===============")
-        player.loop && window.clearInterval(player.loop)
-        player.pause()
-        player.endAudio()
-        Module.cwrap('release', 'number', [])()
-        Module.cwrap('initializeDecoder', 'number', [])()
-        player.stream = new Uint8Array()
-        player.frameList = []
-        player.durationMs = -1.0
-        player.videoPTS = -1
-    }
-    player.nextNalu = (onceGetNalCount=1) => {
```

Comment:
Why remove the nextNalu?

It's a decode part of nalu slice (h.265 unit)

I think dont remove the nalu decoder, its very import for ws or stream decode